docs/2.10.0/docs/canton/usermanual/packagemanagement.rst

```diff
@@ -61,235 +61,181 @@ Understanding Package Vetting
-
-This error goes back to the fact that both participants not only upload the DAR, but also publicly declare towards
-their peers that they are ready to receive transactions referring to certain packages. This declaration happens
-automatically when you upload a DAR. The package vettings can be inspected using (preview):
-
-.. snippet:: package_dar_removal
-    .. success(output=8):: participant2.topology.vetted_packages.list()
-
-Vetting is necessary, as otherwise, a malicious participant might send a transaction referring to package a receiver
-does not have, which would make it impossible for the receiver to process the transaction, leading to a ledger fork.
-As transactions are valid only if all involved participants have vetted the used packages, this attack cannot happen.
-
-Removing Packages and DARs
---------------------------
+If the DAR has not been uploaded to one of the involved participants or it has not been vetted, then transaction submissions are rejected with a
+``NO_DOMAIN_FOR_SUBMISSION`` error (with additional metadata listing which package has not been vetted on which participant).
 
 .. note::
+    The unit of Daml code for which a Canton participant can issue topology transactions is the Daml package.
+    However, in practice, to ensure that transactions can be successfully submitted using any package of a specific DAR,
+    the Canton vetting APIs act at the DAR level (i.e. for all of a DAR's packages, transactionally).
+    The APIs in question are the console commands: ``participant.dars.vetting.enable`` and ``participant.dars.vetting.disable`` as well
+    as the Admin API calls: ``PackageService.vetDar`` and ``PackageService.unvetDar``.
 
-    Note that package and DAR removal is under active development. The behaviour described in this documentation may
-    change in the future. Package and DAR removal is a preview feature and should not be used in production.
-
-Canton supports removal of both packages and DARs that are no longer in use. Removing unused packages and DARs has the following advantages:
-
-- Freeing up storage
-
-- Preventing accidental use of the old package / DAR
-
-- Reducing the number of packages / DARs that are trusted and may potentially have to be audited
-
+Vetting happens automatically when you upload a DAR, although it can be disabled by a request flag.
 
+.. snippet:: package_dar_vetting
+    .. success:: participant2.dars.upload("dars/CantonExamples.dar", vetAllPackages = false)
 
-Certain conditions must to be met in order to remove packages or DARs. These conditions are designed to prevent removal of packages or DARs that are currently in use.
-The rest of this page describes the requirements.
+The packages contained in a DAR can also be vetted explicitly when the DAR upload operation was performed without vetting
 
-Removing DARs
-~~~~~~~~~~~~~
+.. snippet:: package_dar_vetting
+    .. success:: participant2.dars.vetting.enable(hash)
```

Comment:
Wondering whether `participant2.dars.vetting.vet(hash)` would be better in terms of consistency. WDYT?

Reply:
Yep, I wanted to change it, but the API exists for some releases now so we would break backwards compatibility. Is that fine?

Reply:
Or we can add a second console command alongside the first one and document only the new one?

Reply:
Ok, I'll consider it after this as it's orthogonal

Reply:
@rgugliel-da Extracted your suggestion in a separate issue tracker for UX improvements on vetting: https://github.com/DACH-NY/canton/issues/23459